src/v/reflection/async_adl.h

```diff
@@ -35,6 +35,41 @@ struct async_adl {
 
 namespace detail {
 
+template<template<class...> class GenericList, class T, class... Args>
+struct async_adl_list {
+    using List = GenericList<T, Args...>;
```

Comment:
do you think it'd be more robust to take the type (e.g. std::vector<T>) as the template parameter, and then derive T from `using element_type = T::value_type`? I don't think it really matters for us, but it strikes me that this is generic in behavior / interface, but assumes a certain ordering to template parameters.

or maybe we can use the https://en.cppreference.com/w/cpp/experimental/ranges/iterator/ForwardIterator concept?

Reply:
Nice catch, this should of been changed this way after I implemented Bens changes